Lib/glyphsLib/classes.py

```diff
@@ -1747,6 +1760,91 @@ def _set_metric(self, metricname, value):
             self.metrics.append(GSMetricValue(position=0))
         self.metrics[metricIndex] = GSMetricValue(position=value)
 
+    @property
+    def alignmentZones(self):
+        # If there are values that are parsed from file format 2 or which are
+        # explicitly set return those
+        if self._alignmentZones is not None:
+            return self._alignmentZones
+
+        if len(self.metrics) == 0:
+            return []
+
+        validZoneMetrics = (
+            "ascender",
+            "cap height",
+            "x-height",
+            "baseline",
+            "descender",
+        )
+
+        zones = []
+        for index, fontMetric in enumerate(self.font.metrics):
+            if fontMetric.type not in validZoneMetrics:
+                continue
+            metric = self.metrics[index]
```

Comment:
I'm a bit confused.. Where are the metrics (or the stems in a similar piece of code below) actually stored? Inside the GSFont.metrics or in this GSFontMaster.metrics?
Why do you have to iterate over the font.metrics if you then take the item from the master's metrics?

Reply:
ah ok it's explained in https://github.com/schriftgestalt/GlyphsSDK/blob/Glyphs3/GlyphsFileFormat/GlyphsFileFormatv3.md#differences-between-version-2